### PR TITLE
Check BIP32 path buffer bytes length is consistent with path length

### PR DESCRIPTION
**Pull request by agrojean-ledger**
This pull request introduces additional validation for BIP32 path parsing to ensure the buffer length matches the expected number of path elements and adds new tests to verify these checks.

**BIP32 Path Validation Improvements:**

* Updated the `parse_bip32_path` function in `xrp_helpers.c` and its declaration in `xrp_helpers.h` to require both the number of elements and the actual buffer length, and added a check to ensure the buffer contains enough bytes for the specified path length. [[1]](diffhunk://#diff-b1e41306ebe09ad319b600ed2864775c6e7bd8ba739a970db418e528a78c2566R155-R167) [[2]](diffhunk://#diff-91dbc0a77cfd4f95ac4d1a6bea1691af93d790445a805f372666cdf2179cea6cR57)

**Testing Enhancements:**

* Added a new test `test_path_buffer_bytes_and_length_consistency` in `functional_test.py` to ensure that inconsistent path buffer lengths are correctly detected and rejected, and improved existing tests to assert failures when invalid paths are accepted.

* Refactored test cases for signing with large or invalid transactions to use context managers and assert on expected failures. [[1]](diffhunk://#diff-4a98a097cfcfb885e03a44845efac79ab0b3145082efebb5e4f4ae5c83a60dfcL35-R37) [[2]](diffhunk://#diff-4a98a097cfcfb885e03a44845efac79ab0b3145082efebb5e4f4ae5c83a60dfcL45-R49)

These changes collectively strengthen input validation for BIP32 paths and improve test coverage for edge cases.

# Checklist

- [x] App update process has been followed 
- [x] Target branch is `develop` 
- [ ] Application version has been bumped

### CODE REVIEW

File: tests/functional_test.py [hunk 1]
```diff
@@ -52,6 +56,20 @@ def test_path_too_long(backend: BackendInterface, navigator: Navigator):
...
+
+def test_path_buffer_bytes_and_length_consistency(backend: BackendInterface, navigator: Navigator):
+    xrp = XRPClient(backend, navigator)
+    path = Bip32Path.build(DEFAULT_PATH)
+    # Manually construct an APDU with inconsistent path length and byte buffer size
+    # The path length indicates 5 elements, but we only provide bytes for 4 elements
+    payload = path[:-4]  # Remove last 4 bytes to create inconsistency
+    try:
+        backend.raise_policy = RaisePolicy.RAISE_ALL_BUT_0x9000
+        rapdu = xrp._exchange(Ins.GET_PUBLIC_KEY, p2=P2.CURVE_SECP256K1, data=payload)
```
[the remark this answers is not in the document]
Comment (agrojean-ledger) [reply tests/functional_test.py]: Fixed.

File: src/apdu/messages/get_public_key.c [hunk 2]
```diff
@@ -75,9 +75,8 @@ void handle_get_public_key(uint8_t p1,
                            uint16_t data_length,
                            volatile unsigned int *flags,
                            volatile unsigned int *tx) {
-    UNUSED(data_length);
-
     uint8_t bip32_path_length = *(data_buffer++);
+    data_length--;
     uint8_t p2_chain = p2 & 0x3Fu;
```
[the remark this answers is not in the document]
Comment (agrojean-ledger) [reply src/apdu/messages/get_public_key.c:84]: Added a length check for data_length < 1. Other test is done in parse_bip32_path as cerberus suggests.

File: src/swap/handle_check_address.c [hunk 3]
```diff
@@ -20,11 +20,17 @@ int handle_check_address(check_address_parameters_t* params) {
         return 0;
     }
 
+    size_t params_length = params->address_parameters_length;
     uint8_t* bip32_path_ptr = params->address_parameters;
     uint8_t bip32_path_length = *(bip32_path_ptr++);
+    params_length--;
     cx_ecfp_public_key_t public_key;
```
[the remark this answers is not in the document]
Comment (agrojean-ledger) [reply src/swap/handle_check_address.c:32]: Added check

File: src/apdu/messages/sign_transaction.c [hunk 4]
```diff
@@ -197,7 +197,7 @@ void handle_first_packet(uint8_t p1,
 
     work_buffer++;
     data_length--;
-    if (!parse_bip32_path(work_buffer, path_length, path_parsed, MAX_BIP32_PATH)) {
+    if (!parse_bip32_path(work_buffer, path_length, data_length, path_parsed, MAX_BIP32_PATH)) {
```
[the remark this answers is not in the document]
Comment (agrojean-ledger) [reply src/apdu/messages/sign_transaction.c:204]: Added check

File: src/xrp/xrp_helpers.h [hunk 5]
```diff
@@ -54,5 +54,6 @@ int xrp_print_amount(uint64_t amount, char *out, size_t outlen);
 
 bool parse_bip32_path(uint8_t *path,
                       size_t path_length,
+                      size_t path_bytes,
                       uint32_t *path_parsed,
                       size_t path_parsed_length);
```
[the remark this answers is not in the document]
Comment (agrojean-ledger) [reply src/xrp/xrp_helpers.h:59]: Done

**Review by cedelavergne-ledger — APPROVED** [review]
